lenskit-core/src/main/java/org/lenskit/LenskitRecommender.java

```diff
@@ -0,0 +1,152 @@
+     * rather use {@link #build(LenskitConfiguration)} or a {@link LenskitRecommenderEngine}.
+     *
+     * @param graph This recommender's configuration graph.
+     */
+    public LenskitRecommender(DAGNode<Component, Dependency> graph) {
+        injector = new StaticInjector(graph);
+    }
+
+    /**
+     * Get a particular component from the recommender session. Generally you
+     * want to use one of the type-specific getters; this method only exists for
+     * specialized applications which need deep access to the recommender
+     * components.
+     *
+     * @param <T> The type of component to get.
+     * @param cls The component class to get.
+     * @return The instance of the specified component.
+     */
+    public <T> T get(Class<T> cls) {
+        try {
+            return injector.getInstance(cls);
+        } catch (InjectionException e) {
+            throw new RuntimeException("error instantiating component", e);
+        }
+    }
+
+    /**
+     * Get a particular qualified component from the recommender session.  Generally you
+     * want to use one of the type-specific getters; this method only exists for
+     * specialized applications which need deep access to the recommender
+     * components.
+     *
+     * @param <T> The type of component to get.
+     * @param qual The qualifying annotation of the component class.
+     * @param cls The component class to get.
+     * @return The instance of the specified component.
+     */
+    public <T> T get(Class<? extends Annotation> qual, Class<T> cls) {
+        try {
+            return injector.getInstance(qual, cls);
+        } catch (InjectionException e) {
+            throw new RuntimeException("error instantiating component", e);
+        }
+    }
+
+    /**
+     * Get a particular qualified component from the recommender session.  Generally you
+     * want to use one of the type-specific getters; this method only exists for
+     * specialized applications which need deep access to the recommender
+     * components.
+     *
+     * @param <T> The type of component to get.
+     * @param qual The qualifying annotation of the component class.
+     * @param cls The component class to get.
+     * @return The instance of the specified component.
+     */
+    public <T> T get(Annotation qual, Class<T> cls) {
+        try {
+            return injector.getInstance(qual, cls);
+        } catch (InjectionException e) {
+            throw new RuntimeException("error instantiating component", e);
+        }
+    }
+
+    @Override
+    public ItemScorer getItemScorer() {
+        return get(ItemScorer.class);
+    }
+
+    @Override
+    public RatingPredictor getRatingPredictor() {
+        return get(RatingPredictor.class);
+    }
+
+    @Override
+    public ItemRecommender getItemRecommender() {
+        return get(ItemRecommender.class);
+    }
+
+    @Override
+    public void close() {
+        // TODO Implement closing
```

Comment:
No. LK3 will require recommenders to be closed, but we do not yet have the infrastructure to actually close them. So the API is in there as a stub, so new code can be written correctly even before closing does anything.